catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchFilterVisitor.java

```diff
@@ -236,29 +236,17 @@ public Object visit(Intersects filter, Object data) {
             // <ogc:Literal>org.geotools.geometry.jts.spatialschema.geometry.primitive.SurfaceImpl@64a7c45e</ogc:Literal>
             Literal literalWrapper = (Literal) filter.getExpression2();
             Object geometryExpression = literalWrapper.getValue();
-
-            StringBuffer geometryWkt = new StringBuffer();
+            WKTWriter wktWriter = new WKTWriter();
 
             if (geometryExpression instanceof SurfaceImpl) {
-                SurfaceImpl polygon = (SurfaceImpl) literalWrapper.evaluate(null);
-
-                Coordinate[] coords = polygon.getJTSGeometry()
-                        .getCoordinates();
-
-                geometryWkt.append("POLYGON((");
-                for (int i = 0; i < coords.length; i++) {
-                    geometryWkt.append(coords[i].x);
-                    geometryWkt.append(" ");
-                    geometryWkt.append(coords[i].y);
-
-                    if (i != (coords.length - 1)) {
-                        geometryWkt.append(",");
-                    }
-                }
-                geometryWkt.append("))");
-                openSearchFilterVisitorObject.setSpatialSearch(new SpatialFilter(geometryWkt.toString()));
-
-                LOGGER.debug("geometryWkt = [{}]", geometryWkt.toString());
+                SurfaceImpl surface = (SurfaceImpl) literalWrapper.evaluate(null);
+                Polygon polygon = (Polygon) surface.getJTSGeometry();
+                openSearchFilterVisitorObject.setSpatialSearch(new SpatialFilter(wktWriter.write(
+                        polygon)));
+            } else if (geometryExpression instanceof Polygon) {
+                Polygon polygon = (Polygon) literalWrapper.evaluate(null);
+                openSearchFilterVisitorObject.setSpatialSearch(new SpatialFilter(wktWriter.write(
+                        polygon)));
```

Comment:
❓ It looks like these visit methods have the exact same logic, would it be worth abstracting?